rtil/src/threads/ue/rebo/livesplit.rs

```diff
@@ -97,4 +126,57 @@ impl Run {
     pub fn attempt_count() -> u32 {
         LIVESPLIT_STATE.lock().unwrap().timer.run().clone().attempt_count()
     }
+    pub fn save_splits(path: impl AsRef<Path>) -> Result<(), SplitsSaveError> {
+        let state = LIVESPLIT_STATE.lock().unwrap();
+        let path = path.as_ref();
+        let filename = path
+            .file_name().expect("Could not get filename")
+            .to_str().expect("Could not convert filename to &str")
+            .to_owned();
+        let splits_path = LiveSplit::splits_path();
+        let parent = match path.is_absolute() {
+            true => path.parent().expect("Could not get parent directory")
+                .to_str().expect("Could not convert parent dir to &str"),
+            false => splits_path
+                .to_str().expect("Could not convert filename to &str"),
+        };
+        let filename = sanitize_filename::sanitize(filename);
+        let path = match path.is_relative() {
+            true => LiveSplit::splits_path().join(&filename),
+            false => path.to_path_buf(),
+        };
+        if parent != LiveSplit::splits_path().to_str().unwrap() && !VALID_SPLITS.lock().unwrap().contains(&path.to_path_buf()) {
+            return Err(SplitsSaveError::DisallowedFilePath(filename, "Disallowed file path".to_owned()));
+        }
+        let file = match File::create(path) {
+            Ok(file) => file,
+            Err(e) => return Err(SplitsSaveError::CreationFailed(filename, e.to_string())),
+        };
+        let writer = BufWriter::new(file);
+        match save_timer(&state.timer, IoWrite(writer)) {
+            Ok(_) => (),
+            Err(e) => return Err(SplitsSaveError::SaveFailed(filename, e.to_string())),
+        };
+        Ok(())
+    }
+    pub fn load_splits(path: impl AsRef<Path>) -> Result<(), SplitsLoadError> {
+        let filename = path
+            .as_ref()
+            .file_name().expect("Could not get filename")
+            .to_str().expect("Could not convert filename to &str")
+            .to_owned();
+        let file = match fs::read(path.as_ref()) {
+            Ok(file) => file,
+            Err(e) => return Err(SplitsLoadError::OpenFailed(filename.to_owned(), e.to_string())),
+        };
+        let parsed_run = match composite::parse(&file, None) {
+            Ok(parsed_run) => parsed_run,
+            Err(e) => return Err(SplitsLoadError::ParseFailed(filename.to_owned(), e.to_string())),
+        };
+        let mut state = LIVESPLIT_STATE.lock().unwrap();
+        state.timer.reset(true);
+        state.timer.set_run(parsed_run.run).unwrap();
+        VALID_SPLITS.lock().unwrap().to_owned().insert(path.as_ref().to_path_buf());
```

Comment:
What does the `to_owned()` do here? AFAIU it just clones the `HashSet` the `insert` is performed on the clone. Thus, the value is never inserted into the actual static `HashSet` instance. Have you tested that this works?